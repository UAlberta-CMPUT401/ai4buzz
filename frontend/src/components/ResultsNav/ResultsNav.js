import React, { useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import styles from './ResultsNav.module.css';

const ResultsNav = ({ children }) => {
  const history = useHistory();
  const location = useLocation();

  const path = location.pathname;

  return (
    <div className="container">
      <div className={styles.navbar}>
        <div
          className={`${styles.navTab} ${
            path === '/results' ? styles.navTabActive : ''
          }`}
          onClick={() => history.push('/results')}
        >
          Results
        </div>
        <div
          className={`${styles.navTab} ${
            path === '/collage' ? styles.navTabActive : ''
          }`}
          onClick={() => history.push('/collage')}
        >
          Collage
        </div>
        <div
          className={`${styles.navTab} ${
            path === '/dendrogram' ? styles.navTabActive : ''
          }`}
          onClick={() => history.push('/dendrogram')}
        >
          Dendrogram
        </div>
      </div>
      {children}
    </div>
  );
};

export default ResultsNav;
