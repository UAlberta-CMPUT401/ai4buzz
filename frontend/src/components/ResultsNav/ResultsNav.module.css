.navbar {
  display: flex;
  width: 100%;
  justify-content: center;
}

.navTab {
  font-size: 1.3rem;
  margin: 0 1rem;
  padding: 1rem;
  color: rgb(161, 161, 161);
  cursor: pointer;
  user-select: none;
}

.navTabActive {
  color: #fff;

  /* background: #ff6ed0;
  background: -webkit-linear-gradient(
    to right,
    #ff6ed0 0%,
    #a52bff 50%,
    #00edff 100%
  );
  background: -moz-linear-gradient(
    to right,
    #ff6ed0 0%,
    #a52bff 50%,
    #00edff 100%
  );
  background: linear-gradient(to right, #ff6ed0 0%, #a52bff 50%, #00edff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent; */
}
