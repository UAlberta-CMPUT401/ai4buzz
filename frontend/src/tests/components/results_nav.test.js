import React from 'react';
import ResultsNav from '../../components/ResultsNav/ResultsNav';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/results',
  }),
}));

it('Renders Results Nav', () => {
  const tree = renderer
    .create(
      <ResultsNav>
        <></>
      </ResultsNav>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
