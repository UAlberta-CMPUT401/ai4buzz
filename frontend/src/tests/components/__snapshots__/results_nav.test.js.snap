// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Results Nav 1`] = `
<div
  className="container"
>
  <div
    className="navbar"
  >
    <div
      className="back"
      onClick={[Function]}
    >
      Back to image upload
    </div>
    <div
      className="navTab navTabActive"
      onClick={[Function]}
    >
      Results
    </div>
    <div
      className="navTab "
      onClick={[Function]}
    >
      Collage
    </div>
    <div
      className="navTab "
      onClick={[Function]}
    >
      Dendrogram
    </div>
    <div
      className="logo"
    >
      AI4Buzz
    </div>
  </div>
</div>
`;
